LoRaEngine/test/LoRaWan.IntegrationTest/C2DMessageTest.cs

```diff
@@ -126,7 +121,7 @@ public async Task Test_OTAA_Confirmed_Receives_C2D_Message_With_RX_Delay_2()
 
                 // check if c2d message was found
                 // 0000000000000009: C2D message: 58
-                var c2dLogMessage = $"{device.DeviceID}: cloud to device message: {this.ToHexString(c2dMessageBody)}";
+                var c2dLogMessage = $"{device.DeviceID}: done completing cloud to device message, id: {c2dMessage.MessageId}";
```

Comment:
I liked that the test would verify the content of the received message, to ensure we are not messing up the deserialization. Was it hard to maintain?